lib/iris/cube.py

```diff
@@ -781,6 +781,27 @@ def __init__(self, data, standard_name=None, long_name=None,
             for cell_measure, dims in cell_measures_and_dims:
                 self.add_cell_measure(cell_measure, dims)
 
+        # When True indexing may result in a view onto the original data array,
+        # to avoid unnecessary copying.
+        self._share_data = False
+
+    @property
+    def share_data(self):
+        """
+        Share cube data when slicing/indexing cube if True.
+        Setting this flag to True will realise the data payload,
+        if it is lazy, as lazy data cannot currently be shared across cubes.
+        """
+        return self._share_data
+
+    @share_data.setter
+    def share_data(self, value):
+        # If value is True: realise the data (if is hasn't already been) as
+        # sharing lazy data is not possible.
+        if value and self.has_lazy_data():
+            _ = self.data
```

Comment:
>> why does it assign the unnecessarily loaded data into _?

> this is a pattern that has been used elsewhere to avoid any risk the the data will be streamed to standard out

Can I have an example of that from existing Iris code, please? Of course, it would be much the better if this method were not loading data at all.